src/GraphQL/Types/Collections/SchemaTypes.cs

```diff
@@ -56,6 +56,61 @@ public class SchemaTypes : IEnumerable<IGraphType>
             [typeof(Uri)] = typeof(UriGraphType),
         });
 
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(GraphQLClrInputTypeReference<>))]
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(GraphQLClrOutputTypeReference<>))]
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(ListGraphType<>))]
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(NonNullGraphType<>))]
+        static SchemaTypes()
+        {
+            // The above attributes preserve those classes when T is a reference type, but not
+            // when T is a value type, which is necessary for GraphQL CLR type references.
```

Comment:
> but not when T is a value type

🤔 This is probably expected, because otherwise the compiler had to create an infinitely large set of native code for all kinds/sizes of structures.